src/components/layout-components/localization-buttons.tsx

```diff
@@ -20,29 +42,30 @@ const LocalizationButtons = () => {
     }
   };
 
-  /**
-   * Renders localization buttons
-   */
-  const renderLocalizationButtons = () => (
-    <ToggleButtonGroup
+  useEffect(() => {
```

Comment:
I could but then the first click wouldn't update because updated states aren't available immediately upon setting, unless I'm missing something. Hence a useEffect to ensure immediate update upon availability.